data_linter/validation.py

```diff
@@ -156,31 +170,198 @@ def match_files_in_land_to_config(config: dict) -> dict:
+                        offset += 1
+                    else:
+                        bins[i] = curr_bin
+                        has_been_binned = True
+                        break
+            if not has_been_binned:
+                bins[i] = curr_bin
+
+        bins[i] = curr_bin
+        bins = [i for i in bins if i != []]
+
+        # create the configs for the given bins
+        for i, packed_bin in enumerate(bins):
+            config_n = deepcopy(config)
+            config_n.pop("tables")
+            config_n["tables"] = {}
+
+            for table in packed_bin:
+                curr_table_name = table.pop("table-name")
+
+                if config_n["tables"].get(curr_table_name):
+                    # it exists, so just add to matched files
+                    config_n["tables"][curr_table_name]["matched_files"].append(
+                        table["file-name"]
+                    )
+                else:
+                    # it doesn't exist, do a full copy of all attributes
+                    mfile = table.pop("file-name")
+                    table.pop("file-size")
+                    config_n["tables"][curr_table_name] = deepcopy(table)
+                    config_n["tables"][curr_table_name]["matched_files"] = []
+                    config_n["tables"][curr_table_name]["matched_files"].append(mfile)
+
+            # upload the config to temp storage, into config
+            with tempfile.NamedTemporaryFile(
+                suffix=".yml", prefix="config_"
+            ) as tmp_file:
+
+                with open(tmp_file.name, "w") as yaml_out:
+                    yaml.dump(config_n, yaml_out, default_flow_style=False)
+
+                tmp_file_name = tmp_file.name.split("/")[-1]
+                local_file_to_s3(tmp_file.name, f"{s3_temp_path}/{i}/{tmp_file_name}")
+
+    else:
+        pass
+        # do local equiv maybe lol
+
+
+def validate_from_chunked_configs(config: dict, config_num: int) -> bool:
+    land_base_path = config["land-base-path"]
+    land_base_path_is_s3 = land_base_path.startswith("s3://")
+
+    if land_base_path_is_s3:
+        s3_temp_path = os.path.join(
+            land_base_path, f"data_linter_temporary_fs/configs/{config_num}"
+        )
+
+        config_file_paths = get_filepaths_from_s3_folder(s3_temp_path)
+        if not config_file_paths:
+            return False
+
+        s3_client = boto3.client("s3")
+
+        all_configs = []
+        for config_file_path in config_file_paths:
+            bucket, key = s3_path_to_bucket_key(config_file_path)
+            config_file_obj = s3_client.get_object(Bucket=bucket, Key=key)
+            all_configs.append(yaml.safe_load(config_file_obj["Body"].read()))
+
+        for config in all_configs:
+            validate_data(config)
+
+        return True
+
+
 def validate_data(config: dict):
 
-    utc_ts = int(datetime.utcnow().timestamp())
+    # set up logging
+    # log, log_stringio = logging_setup()
+
```

Comment:
Was this causing the double logging?

Reply:
yes - I think. the logger had been declared in multiple places